open-widget-framework/test/defaults_test.js

```diff
@@ -0,0 +1,389 @@
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    instance.addWidget()
+
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(1)
+    expect(dummyProps.renderEditWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  it('editWidget causes just renderEditWidgetForm to be called', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    instance.editWidget(dummyId)
+
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderEditWidgetForm.withArgs(dummyId).callCount).to.equal(1)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  it('toggleEditWidget causes renderAddWidgetButton to be called', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    const renderAddWidgetButtonSpy = sinon.spy(instance, 'renderAddWidgetButton')
+    instance.toggleEditMode()
+
+    expect(renderAddWidgetButtonSpy.callCount).to.equal(1)
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderEditWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  // Click event tests: Test that the various buttons cause the appropriate reactions
+  it('edit-widget-list-btn calls toggleEditMode on click', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    const toggleEditModeSpy = sinon.spy(instance, 'toggleEditMode')
+    const btn = mount(instance.render())
+
+    btn.find('#edit-widget-list-btn').simulate('click')
+    expect(toggleEditModeSpy.callCount).to.equal(1)
+  })
+
+  it('add-widget-btn calls addWidget on click', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    const addWidgetSpy = sinon.spy(instance, 'addWidget')
+    const btn = mount(instance.renderAddWidgetButton())
+
+    expect(btn.exists('#add-widget-btn')).to.equal(true)
+    btn.find('#add-widget-btn').simulate('click')
+    expect(addWidgetSpy.callCount).to.equal(1)
+  })
+})
+
+describe('<_defaultLoader />', () => {
+  it('renders a default-loader component', () => {
+    const wrap = mount(<Loader/>)
+
+    expect(wrap.exists('.default-loader')).to.equal(true)
+  })
+})
+
+describe('< _defaultRenderer />', () => {
+  const dummyTitle = 'dummyTitle'
+  const dummyHTML = <p>dummyHTML</p>
+  const dummyProps = {
+    title: dummyTitle,
+    html: 'dummyHTML',
+  }
+
+  it('sets title and html elements on render', () => {
+    const wrap = mount(<Renderer {...dummyProps}/>)
+    const htmlWrap = mount(dummyHTML)
+
+    expect(wrap.find('.widget-title').text()).to.equal(dummyTitle)
+    expect(wrap.find('.widget-text').text()).to.equal(htmlWrap.text())
```

Comment:
Can we change this to compare the HTML instead of the text? We should also change the `html` prop in `dummyProps` to something containing HTML so we can make sure it is being rendered correctly.

Reply:
I put the whole expected div in as `dummyHTML`. We should definitely have HTML in the `html` prop.